docker-build/Dockerfile

```diff
@@ -1,57 +1,28 @@
-FROM        debian 
-
-ARG         gsd
-ENV         GITHUB_STABLE_DIFFUSION $gsd
-
-ARG         rsd
-ENV         REQS $rsd
-
-ARG         cs
-ENV         CONDA_SUBDIR $cs
-
-ENV         PIP_EXISTS_ACTION="w"
-
-# TODO: Optimize image size
-
-SHELL       ["/bin/bash", "-c"]
-
-WORKDIR     /
-RUN         apt update && apt upgrade -y \
-            && apt install -y \
-            git \
-            libgl1-mesa-glx \
-            libglib2.0-0 \
-            pip \
-            python3 \
-            && git clone $GITHUB_STABLE_DIFFUSION
-
-# Install Anaconda or Miniconda
-COPY        anaconda.sh .
-RUN         bash anaconda.sh -b -u -p /anaconda && /anaconda/bin/conda init bash
-
-# SD 
-WORKDIR     /stable-diffusion 
-RUN         source ~/.bashrc \
-            && conda create -y --name ldm && conda activate ldm \
-            && conda config --env --set subdir $CONDA_SUBDIR \
-            && pip3 install -r $REQS \   
-            && pip3 install basicsr facexlib realesrgan \          
-            && mkdir models/ldm/stable-diffusion-v1 \
-            && ln -s "/data/sd-v1-4.ckpt" models/ldm/stable-diffusion-v1/model.ckpt 
-
-# Face restoreation
-# by default expected in a sibling directory to stable-diffusion
-WORKDIR     /       
-RUN         git clone https://github.com/TencentARC/GFPGAN.git 
-
-WORKDIR     /GFPGAN 
-RUN         pip3 install -r requirements.txt \
-            && python3 setup.py develop \
-            && ln -s "/data/GFPGANv1.4.pth" experiments/pretrained_models/GFPGANv1.4.pth
-
-WORKDIR     /stable-diffusion 
-RUN         python3 scripts/preload_models.py         
-
-WORKDIR     /
-COPY        entrypoint.sh .
-ENTRYPOINT ["/entrypoint.sh"]
\ No newline at end of file
+FROM nvidia/cuda:11.7.1-base-ubuntu22.04
```

Comment:
Would this Image work without having a NVIDIA GPU as well?

Reply:
I have actually tested without that line entirely and it still works, so it will be removed. A lot of this Dockerflle came from someone else, I'm just slowly modifying it to work better. Also have to implement an option to change requirement files.

Reply:
Now I am confused, you built the image without the base Image?

Reply:
There are 2 bases, cuda and miniconda3, miniconda3 on its own allowed a working image to be built.

Reply:
This is rly confusing to me... Why change the image than at all when you say it works without using the nvidia image?!

Did you try to build the existing image like it is [described here](https://invoke-ai.github.io/InvokeAI/installation/INSTALL_DOCKER/)?

Reply:
Because it was pull from this, [#969](https://github.com/invoke-ai/InvokeAI/discussions/969) I don't know why it was done. And yes I did it was detailed in a discord conversation here. https://discord.com/channels/1020123559063990373/1029545193998921738 to describe it briefly that one @hipsterusername contained a lot of old info.

Reply:
Ah - totally ovesaw that Line two is just the next Base Image, without haveing done anything with the first one, so this is only causing other people to have to pull and build this image and then not even use it ;P

Reply:
Yeah, that's what I notice, I never saw someone do it before so I just assumed it had a purpose. When I started changing it to work with AMD cards also, removing it just worked. So yeah removing it in my opinion is the best option.

Reply:
I still don't understand the problem with the existing image, which is way cleaner 🤷🏻‍♂️

Reply:
Let me re-run through the steps so I can grab the output. I can work on changing the existing image to work if that would be better.

Reply:
@mauwii do you have a discord handle? I assume dumping large error messages here would be unpreferred.

Reply:
So I don't want to be rude or in any way offend you, but imho you should first learn more about docker and best practices, since what I saw yet was not rly the best approach.

But please don't feel offended or discouraged, since I also totaly apreciate that you want to contribute and I am sure after diving a bit deeper you will come up with something great 😎

Reply:
@mauwii - I think the docker image had gotten outdated, so even “non best practices” attempts should be fine :)

Reply:
in the end this is not my decission or my repo, but whom is this docker image designed for? imho it is to host the page on some cloudservice / kubernetes / you-name-it. There it would be absolutley a pain in the ... to have a almost 5GB Model file in it, since this will cause a lot of trouble / traffic while it could always stay the same.

I just tried to help here and point to things which are not in a state where I would merge it in a Repo forked by 5k people, since having a outdated dockerfile replaced by a even more obscure one wouldn't help anybody as well....

Reply:
I know this PR has a lot of issues, that's why it was in a draft state, I didn't even want to suggest merging it, but a centralized place for making a new one would be better than having a ton of people working independently on it. And at this point this docker file is more for development purposes and hosting locally using docker. The system has no username/password system so putting it in the cloud would be a bad idea to begin with, but I digress. I'm going to make the changes I need to get it to work on my machine first since I can't reliably test changes on something that doesn't work at all. If in the end the change isn't wanted or needed then the PR can just be closed.

Reply:
For something utilizing the GPU I would always prefer to run it native to have 100% of my resources available to get the job done as quick as possible, while with docker it would take way longer since running isolated from the system.

And of course you need have some kind of authorization when running a container in the cloud, to authorize f.e. between the container and a database, so there are always things like Tokens, Name/PW, Certificates, ... necesarry, but bakeing it into the image is definitly the worst option, you would mount it, add it as env, or whatever.

And as I say, I totally apreciate @ookami125 trying to come up with something and contribute to the project vs about 4500+ people who forked this repo but only use the tool (which is fine and one of the reasons for open source projects, but contributing is like trying to help all those people which which honors the person(s) doing this somehow).

So keep up your enthusiasm @ookami125 and keep on making the world a better place for everybody :octocat: